.gitlab-ci.yml

```diff
@@ -52,7 +52,7 @@ stages:
     - *run_update_uberenv
     - EXEC_PREFIX="srun ${SLURM_ACCOUNT}"
     #DEPENDENCIES
-    - RESOURCES="-t 30 -N 1"
+    - RESOURCES="-t 45 -N 1"
```

Comment:
We cannot ask more than 30 min for a pdebug allocation (see `SLURM_ACCOUNT` value).
But pdebug was critical for `pr_workflow`. Since dependencies are built only in `main_workflow`, we could change `SLURM_ACCOUNT` not to use pdebug in that case.

My suggestion:

```
.pr_workflow:
  variables:
    SLURM_ACCOUNT: -p pdebug -A eng
    LSF_ACCOUNT: -q pdebug -G eng
  rules:
    - if: '$CI_COMMIT_BRANCH != "main" && $CI_COMMIT_BRANCH != "develop" && $BUILD_DEPS != "ON"' #run only if ...

.main_workflow:
  variables:
    SLURM_ACCOUNT: -A eng
    LSF_ACCOUNT: -G eng
  rules:
    - if: '$CI_COMMIT_BRANCH == "main" || $CI_COMMIT_BRANCH == "develop" || $BUILD_DEPS == "ON"' #run only if ...
```

Reply:
Since this is on quartz, I think we have up to 60 minutes:
```
 Jobs are scheduled per node. Quartz has 2 scheduling pools (or partitions) :
    pdebug      46
    pbatch -  2558 nodes (92088 cores), batch use only.
   
   Pools               Max nodes/job       Max runtime
   ---------------------------------------------------
   pdebug                    8(*)           60 minutes
```

That being said, this change may become necessary if/when more TPLs are added.

Reply:
Ah ah! That will teach me to comment without verifying :)

Reply:
Yeah, I think we might hit the limit with PETSc. I'll add an issue on this.